trio/_abc.py

```diff
@@ -254,10 +284,10 @@ async def aclose(self):
 
         """
 
-    async def __aenter__(self):
+    async def __aenter__(self: _T) -> _T:
         return self
 
-    async def __aexit__(self, *args):
+    async def __aexit__(self, *args: object) -> None:
```

Comment:
shouldn't that be
```python
async def __aexit__(self, exc_type: Optional[Type[BaseException]],
        exc_value: Optional[BaseException],
        exc_trackback: Optional[TracebackType]) -> Optional[bool]:
```
As that is the type the contextmanager protocol uses
(Maybe with added `= None`s, not sure about that)

Reply:
I don't think this helps anything, but it does seem more proper.  Though I think `-> None` is an accurate and good expression of the behavior.  Thanks.

Added to the list in the OP.

Reply:
using `-> None` would disallow subclasses from returning `True` if they have some reason for that
(returning `None` and `False` have the same effect)

Reply:
Good point.  Thanks for following up.